py2many/inference.py

```diff
@@ -521,3 +521,22 @@ def visit_Subscript(self, node):
                     node.annotation.lifetime = definition.annotation.lifetime
         self.generic_visit(node)
         return node
+
+    def visit_For(self, node):
+        self.visit(node.target)
+        self.visit(node.iter)
+        if (
+            isinstance(node.target, (ast.Name, ast.Attribute))
+            and hasattr(node.iter, "annotation")
+            and isinstance(node.iter.annotation, ast.Subscript)
+        ):
+            typ = node.iter.annotation.slice
+            # ast.Subscript.slice might be wrapped in python < 3.9 with an ast.Index object,
```

Comment:
Does `self._clike._slice_value()` work here, so we have this logic in one place?

Reply:
Fixed.